File: AM_Program/AM_Shared/ExceptionClasses.vb
'*********************************************************************************************************
' Written by Dave Clark for the US Department of Energy 
' Pacific Northwest National Laboratory, Richland, WA
' Copyright 2006, Battelle Memorial Institute
' Created 06/07/2006
'
'*********************************************************************************************************

Option Strict On

Public Class AMFileNotFoundException
	Inherits ApplicationException

	'*********************************************************************************************************
	'Specialized handler for "file not found" exception
	'*********************************************************************************************************

#Region "Module variables"
	Private m_FileName As String
#End Region

#Region "Properties"
	Public ReadOnly Property FileName() As String
		Get
			Return m_FileName
		End Get
	End Property
#End Region

#Region "Methods"
	''' <summary>
	''' Constructor
	''' </summary>
	''' <param name="FileName">Name of file being processed when exception occurred</param>
	''' <param name="Message">Message to be returned in exception</param>
	''' <remarks></remarks>
	Public Sub New(ByVal FileName As String, ByVal Message As String)

		MyBase.New(Message)
		m_FileName = FileName

	End Sub
#End Region

End Class


Public Class AMFolderNotFoundException
	Inherits ApplicationException

	'*********************************************************************************************************
	'Specialized handler for "folder not found" exception
	'*********************************************************************************************************

#Region "Module variables"
	Private m_FolderName As String
#End Region

#Region "Properties"
	Public ReadOnly Property FolderName() As String
		Get
			Return m_FolderName
		End Get
	End Property
#End Region

#Region "Methods"
	''' <summary>
	''' Constructor
	''' </summary>
	''' <param name="FolderName">Name of unfound folder</param>
	''' <param name="Message">Message for exception to return</param>
	''' <remarks></remarks>
	Public Sub New(ByVal FolderName As String, ByVal Message As String)

		MyBase.New(Message)
		m_FolderName = FolderName

	End Sub
#End Region

End Class

Public Class AMFileNotDeletedAfterRetryException
	Inherits ApplicationException

	'*********************************************************************************************************
	'Specialized handler for file deletion exception after multiple retries
	'*********************************************************************************************************

#Region "Enums"
	Public Enum RetryExceptionType
		IO_Exception
		Unauthorized_Access_Exception
	End Enum
#End Region

#Region "Module variables"
	Private m_FileName As String
	Private m_ExceptionType As RetryExceptionType
#End Region

#Region "Properties"
	Public ReadOnly Property FileName() As String
		Get
			Return m_FileName
		End Get
	End Property

	Public ReadOnly Property ExcType() As RetryExceptionType
		Get
			Return m_ExceptionType
		End Get
	End Property
#End Region

#Region "Methods"
	''' <summary>
	''' Constructor
	''' </summary>
	''' <param name="FileName">Name of file causing exception</param>
	''' <param name="ExceptionType">Exception type</param>
	''' <param name="Message">Message to be returned by exception</param>
	''' <remarks></remarks>
	Public Sub New(ByVal FileName As String, ByVal ExceptionType As RetryExceptionType, ByVal Message As String)

		MyBase.New(Message)
		m_FileName = FileName
		m_ExceptionType = ExceptionType

	End Sub
#End Region

End Class

Public Class AMFileNotDeletedException
	Inherits ApplicationException

	'*********************************************************************************************************
	'Specialized handler for file deletion exception
	'*********************************************************************************************************

#Region "Module variables"
	Private m_FileName As String
#End Region

#Region "Properties"
	Public ReadOnly Property FileName() As String
		Get
			Return m_FileName
		End Get
	End Property
#End Region

#Region "Methods"
	''' <summary>
	''' Constructor
	''' </summary>
	''' <param name="FileName">Name of file causing exception</param>
	''' <param name="Message">Message to be returned by exception</param>
	''' <remarks></remarks>
	Public Sub New(ByVal FileName As String, ByVal Message As String)

		MyBase.New(Message)
		m_FileName = FileName

	End Sub
#End Region

End Class

File: AM_Program/AM_Shared/ExceptionClasses.cs

using System;

//*********************************************************************************************************
// Written by Dave Clark and Matthew Monroe for the US Department of Energy
// Pacific Northwest National Laboratory, Richland, WA
// Copyright 2006, Battelle Memorial Institute
// Created 06/07/2006
//
//*********************************************************************************************************

namespace AnalysisManagerBase
{
    /// <summary>
    /// Specialized handler for "file not found" exception
    /// </summary>
    public class AMFileNotFoundException : ApplicationException
    {
        #region "Properties"

        /// <summary>
        /// Path to the file that was not found
        /// </summary>
        public string FilePath { get; }

        #endregion

        #region "Methods"

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="filePath">Path of file not found</param>
        /// <param name="errorMessage">Exception message</param>
        /// <remarks></remarks>
        public AMFileNotFoundException(string filePath, string errorMessage) : base(errorMessage)
        {
            FilePath = filePath;
        }

        #endregion

    }

    /// <summary>
    /// Specialized handler for "folder not found" exception
    /// </summary>
    public class AMFolderNotFoundException : ApplicationException
    {
        #region "Properties"

        /// <summary>
        /// Path to the directory that was not found
        /// </summary>
        public string FolderPath { get; }

        #endregion

        #region "Methods"

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="folderPath">Path of folder not found</param>
        /// <param name="errorMessage">Exception message</param>
        /// <remarks></remarks>
        public AMFolderNotFoundException(string folderPath, string errorMessage) : base(errorMessage)
        {
            FolderPath = folderPath;
        }
        #endregion

    }

    /// <summary>
    /// Specialized handler for file deletion exception after multiple retries
    /// </summary>
    public class AMFileNotDeletedAfterRetryException : ApplicationException
    {
        #region "Enums"

        /// <summary>
        /// Enum for reason that file could not be deleted
        /// </summary>
        public enum RetryExceptionType
        {
            /// <summary>
            /// I/O exception
            /// </summary>
            IO_Exception,

            /// <summary>
            /// Unauthorized access exception
            /// </summary>
            Unauthorized_Access_Exception
        }

        #endregion

        #region "Properties"

        /// <summary>
        /// Path to the file that could not be deleted
        /// </summary>
        public string FilePath { get; }

        /// <summary>
        /// Reason that file could not be deleted
        /// </summary>
        public RetryExceptionType ExcType { get; }

        #endregion

        #region "Methods"

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="filePath">Path of file that could not be deleted</param>
        /// <param name="ExceptionType">Exception type</param>
        /// <param name="errorMessage">Exception message</param>
        /// <remarks></remarks>
        public AMFileNotDeletedAfterRetryException(string filePath, RetryExceptionType ExceptionType, string errorMessage) : base(errorMessage)
        {
            FilePath = filePath;
            ExcType = ExceptionType;
        }
        #endregion

    }

    /// <summary>
    /// Specialized handler for file deletion exception
    /// </summary>
    public class AMFileNotDeletedException : ApplicationException
    {
        #region "Properties"

        /// <summary>
        /// Path to the file that could not be deleted
        /// </summary>
        public string FilePath { get; }

        #endregion

        #region "Methods"

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="filePath">Path of file that could not be deleted</param>
        /// <param name="errorMessage">Exception message</param>
        /// <remarks></remarks>
        public AMFileNotDeletedException(string filePath, string errorMessage) : base(errorMessage)
        {
            FilePath = filePath;
        }
        #endregion

    }
}